Python/chapter01/1.4 - PalinPerm/miguel_1.4_sol.py

```diff
@@ -0,0 +1,114 @@
+"""
+Python version 3.7.0
+1.4 - Palindrome Permutation
+Given a string, write a function to check if it is a permutation of a palindrome.
+A palindrome is a word or phrase that is the same forwards and backwards.  A permutation is a
+rearrangement of letters.  The palindrome does not need to be limited to just dictionary words.
+"""
+import unittest
+import itertools as it
+import collections
+from typing import Callable
+
+
+def is_permutation_of_palindrome(word: str) -> bool:
+    """
+    Checks if a word is a palindrome by analyzing the frequencies of characters.
+    A palindrome is a word or phrase that is the same forwards and backwards.
+    When determining character frequencies, word will be put to lowercase
+    and spaces will not be counted
+    A palindrome also has the following property:
+    * all characters have an even count or all except one are even
+    Runtime: O(n)
+    Space Complexity: O(n)
+    :param word: the word we check, 'word' is a possible permutation of a palindrome
+    :return: true if word is a permutation of a palindrome, false otherwise
+    """
+    char_frequencies = collections.Counter(word.lower().replace(' ', ''))
+    num_odd_freq_chars = 0
+
+    for key, val in char_frequencies.items():
+        if num_odd_freq_chars > 1:
```

Comment:
you are checking this counter every single time, even though you have not actually updated it.